### PR TITLE
Add mdformat to precommit

### PR DESCRIPTION
**Pull request by SorenSpicknall**
# Description

Automatically lint changed markdown files in the repo during precommit. Will supersede #2897 if accepted. This PR includes an initial repo-wide run of the new tool with some small hand fixes (specifically, Kubernetes -> architecture and Kubernetes -> deployment in the Developers section of the docs site had some empty headers that needed to be filled in).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation

## How has this been tested?

Testing has already taken place locally, and will take place on this PR while open.

## Post-merge follow-ups

- [x] No action required
- [ ] Actions required (specified below)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -60,3 +60,13 @@ repos:
     rev: 0.6.1
     hooks:
       - id: nbstripout
+  - repo: https://github.com/executablebooks/mdformat
+    rev: 0.7.16
+    hooks:
+    - id: mdformat
+      exclude: ^(warehouse/analyses/|warehouse/macros/|warehouse/models/|warehouse/secrets/|warehouse/seeds/|warehouse/snapshots/|warehouse/tests/)
```
Comment (SorenSpicknall) [inline .pre-commit-config.yaml:67]: NOTE: this specific exclude pattern was put in place so that `warehouse/README.md` would continue to be linted

Comment (atvaccaro) [reply .pre-commit-config.yaml:67]: a negative lookahead could be more straightfoward `^warehouse/(?!README.md)` and would also cover future folders

**Review by atvaccaro — APPROVED** [review]
lgtm except for one regex recommendation

I'm not sure of a good way to test this except by manually comparing all of the rendered pages :(

File: .pre-commit-config.yaml (hunk 1 above)
Comment (SorenSpicknall) [reply .pre-commit-config.yaml]: Resolved in [06c9957](https://github.com/cal-itp/data-infra/pull/2906/commits/06c9957cd4d123473f8cd621857a5dd524c9dfed) and tested locally by reverting the linting changes from the previous commit and running pre-commit to see which files the new exclusions acted on.

**Comment by SorenSpicknall** [comment]
Closing in favor of #2923

**Pull request closed without merging**